### PR TITLE
remove unused oidc-proxy-client and other deprecated stuff

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
Since we updated the way our e2e tests communicate with Dex in #4886, the oidc-proxy-client was only left for upgrade tests. Since we removed those in favor of the Kubermatic Installer, it's now finally time to delete the unused code.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/hack/ci/testdata/oauth_values.yaml [hunk 1]
```diff
@@ -15,24 +15,12 @@ dex:
   clients:
   - id: kubermatic
     name: Kubermatic
-    # change this to a new random secret after 2.13 is released
-    secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+    secret: BDZleMTgqON5kiJybIBZM4Si
```
Comment (xrstf) [inline api/hack/ci/testdata/oauth_values.yaml:18]: @zreigz Not sure if this needs to be updated in the dashboard as well. We deploy our own Dex, whereas the dashboard uses dev.kubermatic.io, so maybe it's independent?

Comment (zreigz) [reply api/hack/ci/testdata/oauth_values.yaml:18]: @floreks WDYT?

Comment (floreks) [reply api/hack/ci/testdata/oauth_values.yaml:18]: Dashboard uses dev only for local e2e test, when we are writing them and want to test locally. CI also uses Kubermatic in kind. I don't remember right now what impact it had on the final installation.

**Pull request merged**